docs/guides/nodejs/nitric-and-pgsql.mdx

```diff
@@ -0,0 +1,217 @@
+  migrations: 'file://migrations/todos',
+})
+
+let client: Client
+
+const getClient = async () => {
+  if (!client) {
+    const connectionString = await db.connectionString()
+    client = new Client({ connectionString })
+    await client.connect()
+  }
+  return client
+}
+
+export default getClient
+```
+
+## Database Migrations
+
+Create a migration file to define the `todos` table.
+
+```sql title: migrations/todos/1_create_table.up.sql
+CREATE TABLE IF NOT EXISTS todos (
+    id SERIAL PRIMARY KEY,
+    text TEXT NOT NULL,
+    done BOOLEAN DEFAULT FALSE
+);
+```
+
+## Developing the API
+
+### Building the API
+
+Delete the `services/hello.ts` file and create a new one called `todos.ts`, which will contain our API endpoints.
+
+```ts title: services/todos.ts
+import { api } from '@nitric/sdk'
+import getClient from '../resources/db'
+
+const mainApi = api('main')
+
+// Fetch all todos
+mainApi.get('/todos', async (ctx) => {
+  const client = await getClient()
+  const result = await client.query('SELECT * FROM todos ORDER BY id ASC')
+  return ctx.res.json(result.rows)
+})
+
+// Insert a new todo
+mainApi.post('/todos/:id', async (ctx) => {
+  const { id } = ctx.req.params
+  const { text } = ctx.req.json()
+  const client = await getClient()
+  await client.query('INSERT INTO todos (id, text, done) VALUES ($1, $2, $3)', [
+    parseInt(id),
+    text,
+    false,
+  ])
+})
+
+// Update text of a todo
+mainApi.patch('/todos/:id', async (ctx) => {
+  const { id } = ctx.req.params
+  const { text } = ctx.req.json()
+  const client = await getClient()
+  await client.query('UPDATE todos SET text = $1 WHERE id = $2', [
+    text,
+    parseInt(id),
+  ])
+})
+
+// Toggle todo completion status
+mainApi.patch('/todos/:id/toggle', async (ctx) => {
+  const { id } = ctx.req.params
+  const client = await getClient()
+  const todo = await client.query('SELECT done FROM todos WHERE id = $1', [
+    parseInt(id),
+  ])
+
+  if (todo.rows.length > 0) {
+    const newDone = !todo.rows[0].done
+    await client.query('UPDATE todos SET done = $1 WHERE id = $2', [
+      newDone,
+      parseInt(id),
+    ])
+  }
+})
+
+// Delete a todo
+mainApi.delete('/todos/:id', async (ctx) => {
+  const { id } = ctx.req.params
+  const client = await getClient()
+  await client.query('DELETE FROM todos WHERE id = $1', [parseInt(id)])
+})
+```
+
+### Start testing with `nitric start`
+
+Run `nitric start` to start your local database and APIs.
+
+```bash
+nitric start
+```
+
+Using the dashboard, apply your migrations from the databases tab to initialize your local db.
+
+![Apply migrations](/docs/images/guides/nitric-and-pgsql/migrations.png)
+
+### Add some todos using the Nitric dashboard
+
+Open the local dashboard at <a target="_blank" href="http://localhost:49152">localhost:49152</a>, then navigate to the `POST /todos/{id}` endpoint and fill in an `id` as the path param.
+
+![Add id param for creating a todo](/docs/images/guides/nitric-and-pgsql/step-1.png)
```

Comment:
This needs to be updated with a new screenshot, the ID shouldn't be provided when the todo item is first created, it's autogenerated